core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -274,25 +277,28 @@ private[spark] object JettyUtils extends Logging {
       conf: SparkConf,
       serverName: String = ""): ServerInfo = {
 
-    val collection = new ContextHandlerCollection
     addFilters(handlers, conf)
 
     val gzipHandlers = handlers.map { h =>
+      h.setVirtualHosts(Array("@" + SPARK_CONNECTOR_NAME))
```

Comment:
I don't understand what you mean. Without this change the UI does not work at all. The test I added already covers it.